lean4-infoview/src/infoview/interactiveCode.tsx

```diff
@@ -68,18 +69,43 @@ function TypePopupContents({pos, info, redrawTooltip}: {pos: DocumentPosition, i
   } else return <>Loading..</>
 }
 
-/** Tags in code represent values which can be hovered over to display extra info. */
-function InteractiveCodeTag({pos, tag: ct, fmt}: InteractiveTagProps<CodeToken>) {
+/** Tagged spans can be hovered over to display extra info stored in the associated `SubexprInfo`. */
+function InteractiveCodeTag({pos, tag: ct, fmt}: InteractiveTagProps<SubexprInfo>) {
   const mkTooltip = React.useCallback((redrawTooltip: () => void) =>
     <div className="font-code tl pre-wrap">
       <TypePopupContents pos={pos} info={ct.info}
         redrawTooltip={redrawTooltip} />
     </div>, [pos.uri, pos.line, pos.character, ct.info])
+
+  // We mimick the VSCode ctrl-hover and ctrl-click UI for go-to-definition
+  const rs = React.useContext(RpcContext)
+  const ec = React.useContext(EditorContext);
+  const [goToLoc, setGoToLoc] = React.useState<Location | undefined>(undefined)
+  const [hoverState, setHoverState] = React.useState<HoverState>('off')
+
   return (
-    <WithTooltipOnHover tooltipContent={mkTooltip}>
-      <HighlightOnHoverSpan>
+    <WithTooltipOnHover mkTooltipContent={mkTooltip} onClick={(e, next) => {
+      // On ctrl-click, if location is known, go to it in the editor
+      if (e.ctrlKey && goToLoc !== undefined) void ec.revealPosition({ uri: goToLoc.uri, ...goToLoc.range.start })
```

Comment:
This works, but only if I move the mouse into the token AFTER holding control key.  Can you also add an explicit keydown event handler so it works consistently with how it works in the text editor ?

Reply:
Good point!

Reply:
Done. I didn't want the overhead of every single subexpression registering an event handler, so I made it only react when clicked. The UI is almost the same, with the minor difference that the underline doesn't show up when ctrl-clicked but not ctrl-hovered.